modules/core/02-client/migrations/v7/genesis_test.go

```diff
@@ -0,0 +1,134 @@
+			Sequence: clientState.Sequence,
+			ConsensusState: &v7.ConsensusState{
+				PublicKey:   clientState.ConsensusState.PublicKey,
+				Diversifier: clientState.ConsensusState.Diversifier,
+				Timestamp:   clientState.ConsensusState.Timestamp,
+			},
+			AllowUpdateAfterProposal: true,
+		}
+
+		// set client state
+		any, err := codectypes.NewAnyWithValue(legacyClientState)
+		suite.Require().NoError(err)
+		suite.Require().NotNil(any)
+
+		clients = append(clients, types.IdentifiedClientState{
+			ClientId:    sm.ClientID,
+			ClientState: any,
+		})
+
+		// set in store for ease of determining expected genesis
+		clientStore := suite.chainA.App.GetIBCKeeper().ClientKeeper.ClientStore(suite.chainA.GetContext(), sm.ClientID)
+		bz, err := suite.chainA.App.AppCodec().MarshalInterface(legacyClientState)
+		suite.Require().NoError(err)
+		clientStore.Set(host.ClientStateKey(), bz)
+
+		any, err = codectypes.NewAnyWithValue(legacyClientState.ConsensusState)
+		suite.Require().NoError(err)
+		suite.Require().NotNil(any)
+
+		// obtain marshalled bytes to set in client store
+		bz, err = suite.chainA.App.AppCodec().MarshalInterface(legacyClientState.ConsensusState)
+		suite.Require().NoError(err)
+
+		var consensusStates []types.ConsensusStateWithHeight
+
+		// set consensus states in store and genesis
+		for i := uint64(0); i < numCreations; i++ {
+			height := types.NewHeight(1, i)
+			clientStore.Set(host.ConsensusStateKey(height), bz)
+			consensusStates = append(consensusStates, types.ConsensusStateWithHeight{
+				Height:         height,
+				ConsensusState: any,
+			})
+		}
+
+		clientGenState.ClientsConsensus = append(clientGenState.ClientsConsensus, types.ClientConsensusStates{
+			ClientId:        sm.ClientID,
+			ConsensusStates: consensusStates,
+		})
+	}
+
+	// solo machine clients must come before tendermint in expected
+	clientGenState.Clients = append(clients, clientGenState.Clients...)
+
+	// migrate store get expected genesis
+	// store migration and genesis migration should produce identical results
+	// NOTE: tendermint clients are not pruned in genesis so the test should not have expired tendermint clients
+	err := v7.MigrateStore(suite.chainA.GetContext(), suite.chainA.GetSimApp().GetKey(host.StoreKey), suite.chainA.App.AppCodec())
+	suite.Require().NoError(err)
+	expectedClientGenState := ibcclient.ExportGenesis(suite.chainA.GetContext(), suite.chainA.App.GetIBCKeeper().ClientKeeper)
+
+	cdc, ok := suite.chainA.App.AppCodec().(codec.ProtoCodecMarshaler)
+	suite.Require().True(ok)
+
+	migrated, err := v7.MigrateGenesis(&clientGenState, cdc)
+	suite.Require().NoError(err)
+
+	bz, err := cdc.MarshalJSON(&expectedClientGenState)
+	suite.Require().NoError(err)
+
+	// Indent the JSON bz correctly.
+	var jsonObj map[string]interface{}
+	err = json.Unmarshal(bz, &jsonObj)
+	suite.Require().NoError(err)
+	expectedIndentedBz, err := json.MarshalIndent(jsonObj, "", "\t")
+	suite.Require().NoError(err)
```

Comment:
this actually seems like a useful utility function that would be nice to have any time we are doing a json comparison. WDYT?

Reply:
I could see that. Do we do json comparison anywhere else? Genesis migrations are quite the special case

Reply:
yeah I think it might just be a special case here, I'm happy if you'd prefer to hold out on needing it an additional time before we implement a helper 👍

Reply:
Yea lets go with that. We at least know where to find the code :wink: